lib/kitchen/directions/bake_chapter_references/v1.rb

```diff
@@ -4,28 +4,51 @@ module Kitchen::Directions::BakeChapterReferences
   class V1
     renderable
 
-    def bake(chapter:, metadata_source:, uuid_prefix: '.', klass: 'references')
+    def bake(
+      chapter:,
+      metadata_source:,
+      uuid_prefix: '.',
+      klass: 'references',
+      from_introduction: false
```

Comment:
Assuming that we do always want to bake references in introductions if they are there (discussion[https://openstax.slack.com/archives/C022NETP39N/p1623768035286400]), let's remove this flag and then change the block of code starting on line 21 (from `chapter.non_introduction_pages...`) to

```ruby
chapter.pages.each do |page|
  references = page.references        
  next if references.none?

  if page.is_introduction?
    title =
      <<~HTML
        <a href="##{page.title.id}">
          <h3 data-type="document-title" id="#{page.title.copied_id}">
            <span class="os-text" data-type="" itemprop="">#{page.title_text}</span>
          </h3>
        </a>
      HTML
  else
    references.search('h3').trash        
    title = Kitchen::Directions::EocSectionTitleLinkSnippet.v1(page: page)        
  end

  references.each do |reference|          
    reference.prepend(child: title)        
  end
end
```

Then we can remove the block starting on line 32.

Reply:
```ruby
def bake
  ...
  bake_page_references(page: chapter.introduction_page)

  chapter.non_introduction_pages.each do |page|
    bake_page_references(page: page)
  end
  ...
end


def bake_page_references(page:)
  return if page.nil?

  references = page.references        
  return if references.none?

  if page.is_introduction?
    title =
      <<~HTML
        <a href="##{page.title.id}">
          <h3 data-type="document-title" id="#{page.title.copied_id}">
            <span class="os-text" data-type="" itemprop="">#{page.title_text}</span>
          </h3>
        </a>
      HTML
  else
    references.search('h3').trash        
    title = Kitchen::Directions::EocSectionTitleLinkSnippet.v1(page: page)        
  end

  references.each do |reference|          
    reference.prepend(child: title)        
  end
end
```